### PR TITLE
Adds ability to view and create notes from the customer details page

### PR DESCRIPTION
**Pull request by rhonsby**
### Description

The last major feature left for V2 of the customer details page is the ability to view and create notes which we now can do with this commit 🎉

### Screenshots

https://user-images.githubusercontent.com/1361509/115447331-61c85100-a1e6-11eb-8a6b-b862a4f2be5a.mp4

## Checklist

- [x] Everything passes when running `mix test`
- [x] Ran `mix format`
- [x] No frontend compilation warnings

### CODE REVIEW

**Review by reichert621 — CHANGES_REQUESTED** [review]
woohoooo can't wait for this! 👯 

just a few comments, curious to get your thoughts :)

File: assets/src/components/customers/CustomerDetailsNotes.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            height: '100%',
+          }}
+        >
+          <Spinner size={40} />
+        </Flex>
+      );
+    }
+
+    return (
+      <>
```
[the remark this answers is not in the document]
Comment (rhonsby) [reply assets/src/components/customers/CustomerDetailsNotes.tsx:81]: Oh yeah, good catch. There were multiple elements, but I consolidated them under `Box` and forgot to delete this.

File: assets/src/components/customers/CustomerDetailsNotes.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            customerId={customerId}
+            onCreateNote={this.handleCreateNote}
+          />
+
+          <Divider dashed />
+
+          {customerNotes.length === 0 ? (
+            <Empty image={Empty.PRESENTED_IMAGE_SIMPLE} />
+          ) : (
+            <Flex
```
[the remark this answers is not in the document]
Comment (rhonsby) [reply assets/src/components/customers/CustomerDetailsNotes.tsx:93]: No, not anymore!

**Review by reichert621 — APPROVED** [review]
🎉

**Pull request merged**